parser/class_files.py

```diff
@@ -53,5 +85,128 @@ def parse_images(tag, target, dita_soup):
 
+def parse_summary_and_signatures(tag, target, dita_soup,):
+    td = tag.find("td", {"colspan": "6"})
+    table = td.find_parent("table")
+
+    # Create Dita elements
+    dita_summary_thead = dita_soup.new_tag("thead")
+    dita_summary_tbody = dita_soup.new_tag("tbody")
+
+    dita_signatures_thead = dita_soup.new_tag("thead")
+    dita_signatures_tbody = dita_soup.new_tag("tbody")
+
+    dita_summary_tgroup = dita_soup.new_tag("tgroup")
+    dita_summary_tgroup["cols"] = "6"
+
+    dita_signatures_tgroup = dita_soup.new_tag("tgroup")
+    dita_signatures_tgroup["cols"] = "4"
+
+    dita_summary_table = dita_soup.new_tag("table")
+    dita_signatures_table = dita_soup.new_tag("table")
+
+    dita_summary = dita_soup.new_tag("summary")
+    dita_summary["id"] = "summary"
+
+    dita_colspec = dita_soup.new_tag("colspec")
+    dita_signatures = dita_soup.new_tag("signatures")
+    dita_signatures["id"] = "sigantures"
+
+    for tr_count, tr in enumerate(table.find_all("tr")):
+        dita_row = dita_soup.new_tag("row")
+        cells = tr.find_all("td")
+        for idx, td in enumerate(cells):
+            # Append the first and second <tr> elements to the <summary> element
+            dita_entry = dita_soup.new_tag("entry")
+            dita_entry.string = td.text.strip()
+            # if only one cell, do colspan
+            if len(cells) == 1:
+                dita_entry["nameend"] = "col4"
+                dita_entry["namest"] = "col1"
+                dita_entry["align"] = "center"
+                dita_entry["outputclass"] = "table-separator"
+
+            # if two cells, make the second one wider
+            if len(cells) == 2:
+                if idx == 1:
+                    dita_entry["nameend"] = "col4"
+                    dita_entry["namest"] = "col2"
+
+            dita_row.append(dita_entry)
+
+        if tr_count == 0:
+            dita_summary_thead.append(dita_row)
+        elif tr_count == 1:
+            dita_summary_tbody.append(dita_row)
+        elif tr_count == 2:
+            dita_signatures_thead.append(dita_row)
+        else:
+            dita_signatures_tbody.append(dita_row)
+
+    dita_summary_tgroup.append(dita_summary_thead)
+    dita_summary_tgroup.append(dita_summary_tbody)
+    dita_summary_table.append(dita_summary_tgroup)
+    dita_summary.append(dita_summary_table)
+
+    for count in range(4):
+        dita_colspec = dita_soup.new_tag("colspec")
+        dita_colspec["colnum"] = count + 1
+        dita_colspec["colname"] = f"col{count + 1}"
+        dita_signatures_tgroup.append(dita_colspec)
+
+    dita_signatures_tgroup.append(dita_signatures_thead)
+    dita_signatures_tgroup.append(dita_signatures_tbody)
+    dita_signatures_table.append(dita_signatures_tgroup)
+    dita_signatures.append(dita_signatures_table)
+
+    #Append the summary and the signatures in the dita <body>
+    target.append(dita_summary)
+    target.append(dita_signatures)
+
+    return target
```

Comment:
@AbegaM - we don't process the result of this function, since it adds its content to the `target` object.  I'm pretty sure we can remove the `return` statement.